app/src/main/java/zapsolutions/zap/util/BlockExplorer.java

```diff
@@ -6,123 +6,143 @@
+    }
 
     /**
      * Shows transaction details in a browser window using the preferred block explorer.
      *
      * @param transactionID transaction to show
      * @param ctx
      */
-    public static void showTransaction(String transactionID, Context ctx) {
+    public void showTransaction(String transactionID, Context ctx) {
+        mContext = ctx;
         String explorer = PrefsUtil.getPrefs().getString("blockExplorer", "BlockCypher");
         boolean mainnet = !Wallet.getInstance().isTestnet();
         String networkID = "";
-        String url = "";
-        boolean supported = true;
+        mUrl = "";
+        mSupported = true;
 
         switch (explorer) {
             case "BlockCypher":
                 networkID = mainnet ? "btc" : "btc-testnet";
-                url = "https://live.blockcypher.com/" + networkID + "/tx/" + transactionID;
+                mUrl = "https://live.blockcypher.com/" + networkID + "/tx/" + transactionID;
                 break;
             case "Blockstream":
                 networkID = mainnet ? "" : "testnet/";
-                url = "https://blockstream.info/" + networkID + "tx/" + transactionID;
+                mUrl = "https://blockstream.info/" + networkID + "tx/" + transactionID;
                 break;
             case "Blockstream (v3 Tor)":
                 networkID = mainnet ? "" : "testnet/";
-                url = "http://explorerzydxu5ecjrkwceayqybizmpjjznk5izmitf2modhcusuqlid.onion/" + networkID + "tx/" + transactionID;
+                mUrl = "http://explorerzydxu5ecjrkwceayqybizmpjjznk5izmitf2modhcusuqlid.onion/" + networkID + "tx/" + transactionID;
                 break;
             case "Smartbit":
                 networkID = mainnet ? "www" : "testnet";
-                url = "https://" + networkID + ".smartbit.com.au/tx/" + transactionID;
+                mUrl = "https://" + networkID + ".smartbit.com.au/tx/" + transactionID;
                 break;
             case "Blockchain Reader (Yogh)":
-                supported = mainnet;
-                url = "http://srv1.yogh.io/#tx:id:" + transactionID;
+                mSupported = mainnet;
+                mUrl = "http://srv1.yogh.io/#tx:id:" + transactionID;
                 break;
             case "OXT":
-                supported = mainnet;
-                url = "https://oxt.me/transaction/" + transactionID;
+                mSupported = mainnet;
+                mUrl = "https://oxt.me/transaction/" + transactionID;
                 break;
         }
 
-        if (supported) {
-            // Call the url
-            Intent browserIntent = new Intent(Intent.ACTION_VIEW, Uri.parse(url));
-            ctx.startActivity(browserIntent);
+        if (explorer.equals("Blockstream (v3 Tor)")){
+            startBlockExplorer();
         } else {
-            unsupportedNetwork(explorer, mainnet ? "mainnet" : "testnet", ctx);
+            new UserGuardian(mContext, this).privacyBlockExplorer();
```

Comment:
Just an idea, maybe for another PR.
If we would pass a `Bundle` with extras to the `UserGuardian` class, we could pass the url like
bundle.putExtras(BlockExplorerURL,url)
and receive it in `guardianDialogConfirmed(String DialogName, Bundle extras)` again.
We then could remove the `DialogName` as it could part of the `extras` like
`bundle.putExtras(DialogType,"blockexplorer")`

This is just pseudecode to translate the idea 👍 
Let me know what you think. We could create an issue and tackle it later as well.

Reply:
I actually tried that first. I had the BlockExplorer Class as a static class with static functions.
I was able to transmit the URL with a bundle. But I also needed a Context to start the url from which cannot be included in the bundle. So I ended up with this solution where I thought that is easier.